### PR TITLE
HPCC-13988 CSV Parsing QUOTE char and ESCAPE char cannot be the same

### PR DESCRIPTION
**Pull request by AttilaVamos**
Ignore ESCAPE if it is same as the QUOTE, but comment it into dfu server log

Signed-off-by: Attila Vamos attila.vamos@gmail.com

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman please review.

File: dali/ft/daftformat.cpp [hunk 1]
```diff
@@ -667,12 +667,19 @@ CCsvPartitioner::CCsvPartitioner(const FileFormat & _format) : CInputBasePartiti
...
+    const char * quote = format.quote.get();
+    addActionList(matcher, quote ? quote : "\"", QUOTE, &maxElementLength);
     addActionList(matcher, format.terminate.get() ? format.terminate.get() : "\\n,\\r\\n", TERMINATOR, &maxElementLength);
+
     const char * escape = format.escape.get();
     if (escape && *escape)
-        addActionList(matcher,  escape, ESCAPE, &maxElementLength);
-
+    {
+        if (*escape == *quote)
```
Comment (richardkchapman) [inline dali/ft/daftformat.cpp]: This may core if quote is NULL ?

**Comment by AttilaVamos** [comment]
Theoretically it can core. Code updated. Please re-review.

**Comment by richardkchapman** [comment]
Looks ok - I'm going to defer merging until we know if there will be another rc

**Pull request merged**